src/Lib/Managers/SelectedBoundManager.vala

```diff
@@ -422,4 +420,88 @@ public class Akira.Lib.Managers.SelectedBoundManager : Object {
             snap_manager.populate_decorators_from_data (matches, snap_grid);
         }
     }
+
+    private void scale_from_event (
+        Lib.Items.CanvasItem item,
+        Managers.NobManager.Nob selected_nob,
+        double event_x,
+        double event_y
+    ) {
+        if (!initial_drag_registered) {
+            item.get_transform (out initial_item_transform);
+            initial_drag_registered = true;
+            initial_drag_item_x = item.bounds.x1;
+            initial_drag_item_y = item.bounds.y1;
+            debug ("%f", initial_drag_item_x);
+            debug ("%f", initial_drag_item_y);
+            scale_item_x_adj = 0;
+            scale_item_y_adj = 0;
+
+            // If rotation is multiple of pi / 2 (90), then snap to pixel grid before scale
+            var dummy_x = 1.0;
+            var dummy_y = 1.0;
+            initial_item_transform.transform_distance (ref dummy_x, ref dummy_y);
+            if (Utils.AffineTransform.epsilon_equals_d (1.0, GLib.Math.fabs (dummy_x), double.EPSILON * 10)
+                && Utils.AffineTransform.epsilon_equals_d (1.0, GLib.Math.fabs (dummy_y), double.EPSILON * 10)) {
+                scale_item_x_adj = Utils.AffineTransform.fix_size (initial_drag_item_x) - initial_drag_item_x;
+                scale_item_y_adj = Utils.AffineTransform.fix_size (initial_drag_item_y) - initial_drag_item_y;
+                initial_width = Utils.AffineTransform.fix_size (initial_width);
+                initial_height = Utils.AffineTransform.fix_size (initial_height);
+            }
```

Comment:
This is what causing the shift in the position of the item and the wrong alignment.

Reply:
I think this should be wrapped inside a condition like:
`if (item.rotation != null && item.rotation.rotation != 0) {`
So we avoid running it on item creation, or for those items that don't support rotation, like artboards.

Reply:
rotation != 0 is not the correct condition, since it applies to 90, 180, 270 or 360

Reply:
So, could we do this? 
`if (item.rotation != null && GLib.Math.fmod (item.rotation.rotation, 90) == 0)`